packages/app/components/settings/tabs/account.tsx

```diff
@@ -76,7 +85,45 @@ const ConnectSpotify = ({ redirectUri }: ConnectSocialProps) => {
   );
 };
 
-const { useParam } = createParam<{ did: string; type: string }>();
+const ConnectInstagram = ({ providerId }: { providerId: string }) => {
+  const user = useUser();
+  const isDark = useIsDarkMode();
+
+  const { trigger: disconnectInstagram } = useDisconnectInstagram();
+  const { trigger: addSocial } = useAddMagicSocialAccount();
+
+  return (
+    <View tw="space-between flex-row items-center justify-between py-2 md:py-3.5">
+      <View tw="flex-row items-center">
+        <Instagram height={25} width={25} color={isDark ? "#fff" : "#000"} />
+        <Text tw="ml-2.5 text-base font-medium text-gray-900 dark:text-gray-100">
+          Instagram
+        </Text>
+      </View>
+      <Button
+        variant={
+          user.user?.data.profile.social_login_connections.instagram
+            ? "danger"
+            : "tertiary"
+        }
+        onPress={() => {
+          if (user.user?.data.profile.social_login_connections.instagram) {
+            disconnectInstagram({
+              provider: "instagram",
+              providerId,
+            });
+          } else {
+            addSocial({ type: "instagram" });
+          }
+        }}
+      >
+        {user.user?.data.profile.social_login_connections.instagram
+          ? "Disconnect"
+          : "Connect"}
+      </Button>
+    </View>
+  );
+};
```

Comment:
For the profile, we need the tiny buttons + state (EditProfile.tsx)

And for the connect buttons, the layout should look like so:

![Screenshot 2023-03-12 at 11 55 37 PM](https://user-images.githubusercontent.com/504909/224647811-ddab768a-d7b7-421f-941d-87b7333a88cc.png)

Reply:
FYI: screenshot only one color mode. Inverted colors for dark/light

Reply:
do we need to show the connected state? we show the toast and the router pops.

Reply:
merging this for now. Can do in a follow up PR if required!

Reply:
Yeah, Alex asked for "connected". We can do a follow up PR